test-driver/src/ComparisonContext.ts

```diff
@@ -50,16 +50,31 @@ export class ComparisonContext {
         return fs.readFileSync(path, options);
     }
 
-    setEventStream(ionReader: Reader) {
-        let eventStream = new IonEventStream(ionReader);
+    setEventStream(ionReader: Reader, path: string, args: IonCompareArgs) {
         let events: IonEvent[] = [];
+        try{
+            let eventStream = new IonEventStream(ionReader);
 
-        if(!eventStream.isEventStream) {
-            // processes input stream(text or binary) into event stream
-            events = this.collectInputStreamEvents(events, eventStream);
-            eventStream.events = events;
+            if(!eventStream.isEventStream) {
+                // processes input stream(text or binary) into event stream
+                events = this.collectInputStreamEvents(events, eventStream);
+                eventStream.events = events;
+            }
+            this.eventStream = eventStream;
+        } catch (EventStreamError) {
+            if(EventStreamError.eventstream) {
+                events.push(...EventStreamError.eventstream);
```

Comment:
In the past we've been burned by using `array.push(...someValues)` because it can [run into configured JS VM limits on the number of parameters a function can accept](https://github.com/amzn/ion-js/issues/601).

This is just an FYI; I don't think it's likely to be an issue for `EventStream` since our test cases aren't typically very long.